### DISCUSSION TITLE
Avoid escape characters and print raw data

### DISCUSSION THREAD

**Discussion by veera-chenna** [hairyhenderson/gomplate#1374] [Q&A] [answered]
Hi @hairyhenderson ,

Use case is as follows,
If we take the following json object as datasource,
```{ "test": "no\\escape"}```

When I am trying to print the string, it is escaping the `\\` and printing only `\`,
```Command: gomplate --datasource=test=./test.json?type=application/json -i '{{ range $key, $value := (datasource "test") }}{{ $key }}={{ $value }}{{ end }}```
Output: `test=no\escape`

We wanted to print the raw value of the key, which is `test=no\\escape`

Is there any option in gomplate which we can use to print a string with raw value?

**Comment by hairyhenderson** [comment, marked as answer]
Hi @veera-chenna, I've converted this to a discussion as this doesn't seem to be a bug or feature request.

> When I am trying to print the string, it is escaping the `\\` and printing only `\`,

To the contrary, the JSON string contains a `\` which is escaped as `\\`, so gomplate is behaving correctly by un-escaping it as you output the value.

I think what you want to do is quote the string so that it's re-escaped. There are a few ways you could approach this, but one way is to use `printf` with the `%q` ("quote") formatter:

For example:

```console
$ gomplate -i '{{ $s := "no\\escape" }}{{ printf "%q" $s }}'
"no\\escape"
```

You could also use the [`strings.Quote`](https://docs.gomplate.ca/functions/strings/#strings-quote) function, which is equivalent:

```console
$ gomplate -i '{{ $s := "no\\escape" }}{{ strings.Quote $s }}'
"no\\escape"
```

Also equivalent in this case would be [`data.ToJSON`](https://docs.gomplate.ca/functions/data/#data-tojson):

```console
$ gomplate -i '{{ $s := "no\\escape" }}{{ data.ToJSON $s }}'
"no\\escape"
```

However all of these methods are escaping the string for the purpose of quoting it, so you'll always end up with the string surrounded by double-quotes. If you want the literal raw string `no\\escape`, the correctly-escaped JSON input will need to be `"no\\\\escape"`.

Hope this helps!
